src/widgets/NodeEditor/SceneTabWidget.py

```diff
@@ -1,25 +1,142 @@
+import logging
+
 from PySide6.QtWidgets import QWidget, QGridLayout
+import PySide6
 
-from pyqtgraph.flowchart.Flowchart import Flowchart
+from pyqtgraph.flowchart.Flowchart import Flowchart, Terminal
 from pyqtgraph.flowchart.library import NodeLibrary
 
-from DMXModel import Scene
+from model.board_configuration import Scene, Filter, UniverseFilter
+
+from . import Nodes
 
 class SceneTabWidget(QWidget):
+
+    _type_to_node: dict[int, type] = {
+        0: Nodes.Constants8BitNode.nodeName,
+        1: Nodes.Constants16BitNode.nodeName,
+        2: Nodes.ConstantsFloatNode.nodeName,
+        3: Nodes.ConstantsColorNode.nodeName,
+        4: Nodes.Debug8BitNode.nodeName,
+        5: Nodes.Debug16BitNode.nodeName,
+        6: Nodes.DebugFloatNode.nodeName,
+        7: Nodes.DebugColorNode.nodeName,
+        8: Nodes.Adapters16To8Bit.nodeName,
+        9: Nodes.Adapters16ToBool.nodeName,
+        10: Nodes.ArithmeticsMAC.nodeName,
+        11: Nodes.UniverseNode.nodeName,
+        12: Nodes.ArithmeticsFloatTo16Bit.nodeName,
+        13: Nodes.ArithmeticsFloatTo8Bit.nodeName,
+        14: Nodes.ArithmeticsRound.nodeName,
+        15: Nodes.ColorToRGBNode.nodeName,
+        16: Nodes.ColorToRGBWNode.nodeName,
+        17: Nodes.ColorToRGBWANode.nodeName,
+        18: Nodes.FloatToColorNode.nodeName,
+        19: Nodes.SineNode.nodeName,
+        20: Nodes.CosineNode.nodeName,
+        21: Nodes.TangentNode.nodeName,
+        22: Nodes.ArcsineNode.nodeName,
+        23: Nodes.ArccosineNode.nodeName,
+        24: Nodes.ArctangentNode.nodeName,
+        25: Nodes.SquareWaveNode.nodeName,
+        26: Nodes.TriangleWaveNode.nodeName,
+        27: Nodes.SawtoothWaveNode.nodeName,
+        28: Nodes.LogarithmNode.nodeName,
+        29: Nodes.ExponentialNode.nodeName,
+        30: Nodes.MinimumNode.nodeName,
+        31: Nodes.MaximumNode.nodeName,
+        32: Nodes.TimeNode.nodeName,
+        33: Nodes.SwitchOnDelay8BitNode.nodeName,
+        34: Nodes.SwitchOnDelay16BitNode.nodeName,
+        35: Nodes.SwitchOnDelayFloatNode.nodeName,
+        36: Nodes.SwitchOffDelay8BitNode.nodeName,
+        37: Nodes.SwitchOffDelay16BitNode.nodeName,
+        38: Nodes.SwitchOffDelayFloatNode.nodeName,
```

Comment:
There are currently 44 filters. Some are definitely not implemented yet. I would recommend syncing with main prior to further completing the list tough.